packages/core/src/plugins/html-deserializer/utils/htmlStringToDOMNode.ts

```diff
@@ -1,16 +1,66 @@
 /**
- * Convert HTML string into HTML element.
+ * Convert HTML string into HTML element with custom white-space handling.
  */
 export const htmlStringToDOMNode = (
   rawHtml: string,
-  stripWhitespace = true
+  stripWhitespace = true,
+  preElementBehavior = true
 ) => {
   const node = document.createElement('body');
   node.innerHTML = rawHtml;
-
-  if (stripWhitespace) {
-    node.innerHTML = node.innerHTML.replaceAll(/(\r\n|[\t\n\r])/g, '');
+  const computedStyle = window.getComputedStyle(node);
```

Comment:
I'm not sure checking the computed style of the top-level element is going to work. The HTML to be deserialized may contain a mixture of different whitespace modes.

Reply:
If this doesn't work, you could try walking the DOM tree in a recursive function while keeping track of the current whitespace mode. Then you could transform each text node as you arrive at it according to the rules.